sstables/mx/reader.cc

```diff
@@ -589,8 +563,10 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
                         const sstables::column_translation::column_info& column_info = get_column_info();
                         move_to_next_column();
                         if (_consumer.consume_complex_column_end(column_info) != consumer_m::proceed::yes) {
-                            _state = state::COLUMN;
+                            _state = state::OTHER;
+                            _consuming = false;
```

Comment:
Again, I don't understand or remember what setting _consuming = false was supposed to be doing here and why proceed:no. If this stuff remains in the final version (and it's not temporary code that will be changed in the next patches), I think it deserves a comment.

Reply:
We set _consuming = false here because we have ```goto column_label``` afterwards, where we can reach ```_consumer.consume_row_end()``` without reading any data.

We yield proceed::no, because that's what we got from ```_consumer.consume_complex_column_end(column_info)``` (the ```!= consumer_m::proceed::yes``` is a bit confusing because in other places we check for ```== consumer_m::proceed::no```, I changed it in a later commit)

We could replace the _consuming variable with a new state, though I'm not sure if that would simplify anything.

I think I'll just add a comment next to the _consuming variable declaration